enumerable_methods.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    my_each do |element|
+      next unless yield(element)
+
+      selected << element
+    end
+    selected
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg.nil
+      my_each { |element| return false unless element }
+    else
+      my_each { |element| return false unless check_arg(element, arg) }
+    end
+    true
+  end
+
+  def my_any?(arg = nil, &block)
+    if block_given?
+      my_each { |element| return true if block.call(element) }
+    elsif arg.nil?
```

Comment:
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex

 ```
%w[ant bear cat].any?(/d/)                        #=> false
```